### PR TITLE
tmz 3.1.0 enhancement morse gerke steinheil

### PR DESCRIPTION
**Pull request by t-m-z**
add three tables
- Original Code from Vail
- Gerke
- adapted Steinheil

### CODE REVIEW

File: lib/application/registry.dart
[the remark this answers is not in the document]
Comment (t-m-z) [reply lib/application/registry.dart:1]: I thought we agreed to keep the selections without  searchstrings

[the remark this answers is not in the document]
Comment (Mike-3) [reply lib/application/registry.dart:1]: I mean the symbol tables. The Morse symbol table is already there.

**Comment by t-m-z** [comment]
done

File: lib/tools/crypto_and_encodings/morse/logic/morse.dart [hunk 1]
```diff
@@ -1,47 +1,50 @@
...
-  'CH': '----', '.': '.-.-.-', ',': '--..--', ':': '---...', ';': '-.-.-.', '?': '..--..', '@': '.--.-.',
-  '-': '-....-', '_': '..--.-', '(': '-.--.', ')': '-.--.-', '\'': '.----.', '=': '-...-', '+': '.-.-.', '/': '-..-.',
-  '!': '-.-.--'
+part 'package:gc_wizard/tools/crypto_and_encodings/morse/logic/morse_data.dart';
+
+enum MORSE_CODE {MORSE_ITU, AMERICAN, STEINHEIL, GERKE}
+
+final Map<MORSE_CODE, String> MORSE_CODES = {
+  MORSE_CODE.MORSE_ITU: 'Morse (ITU)',
+  MORSE_CODE.AMERICAN: 'Morse',
```
[the remark this answers is not in the document]
Comment (Mike-3) [reply lib/tools/crypto_and_encodings/morse/logic/morse.dart]: Now I have the same text ("Morse") in the field at the top, no matter which Morse version I select.

[the remark this answers is not in the document]
Comment (t-m-z) [reply lib/tools/crypto_and_encodings/morse/logic/morse.dart]: Changed

File: lib/tools/crypto_and_encodings/morse/widget/morse.dart [hunk 2]
```diff
@@ -98,6 +105,20 @@ class _MorseState extends State<Morse> {
   Widget build(BuildContext context) {
     return Column(
       children: <Widget>[
+        GCWDropDown<MORSE_CODE>(
+          value: _currentCode,
+          items: MORSE_CODES.entries.map((mode) {
+            return GCWDropDownMenuItem(
+                value: mode.key,
+                child: i18n(context, mode.value + '_title'),
```
Comment (Mike-3) [inline lib/tools/crypto_and_encodings/morse/widget/morse.dart:113]: image
Missing item text

**Review by Mike-3 — CHANGES_REQUESTED** [review]

File: lib/tools/crypto_and_encodings/morse/widget/morse.dart (hunk 2 above)
Comment (t-m-z) [reply lib/tools/crypto_and_encodings/morse/widget/morse.dart:113]: corrected

File: lib/tools/crypto_and_encodings/morse/logic/morse.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (Mike-3) [reply lib/tools/crypto_and_encodings/morse/logic/morse.dart]: I also noticed differences in the symbol tables and the selectable buttons in the widget.

Steinheil - no dash in the widget (only 2 * dot)
Morse 1838 - in the symbol table I only see short lines, long lines and dots. There is even more to choose from in the widget
Morse 1844 - I only see a short line in the symbol table. There is even more to choose from in the widget

[the remark this answers is not in the document]
Comment (t-m-z) [reply lib/tools/crypto_and_encodings/morse/logic/morse.dart]: Steinheil: thats correct: Steinheil has no dash - only two dots at different heights
Morse 1838: corrected - was a failure while cleaning to much up
Morse 1844: thats okay: there are three differente dashes: 0, L and T

**Pull request merged**